volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,374 @@
+            daddr_tag = f"{dst_addr}"
+        else:
+            vollog.warning("Unsupported bluetooth protocol '%s'", bt_sock.protocol)
+
+        state = bt_sock.state
+        sock_stat = saddr_tag, daddr_tag, state
+        return bt_sock, sock_stat
+
+class Sockstat(plugins.PluginInterface):
+    """Lists all network connections for all processes."""
+
+    _required_framework_version = (2, 0, 0)
+
+    _version = (2, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(name="kernel", description="Linux kernel",
+                                           architectures=["Intel32", "Intel64"]),
+            requirements.PluginRequirement(name="lsof", plugin=lsof.Lsof, version=(2, 0, 0)),
+            requirements.VersionRequirement(name="linuxutils", component=linux.LinuxUtilities, version=(2, 0, 0)),
+            requirements.BooleanRequirement(name="unix",
+                                            description=("Show UNIX domain Sockets only"),
+                                            default=False,
+                                            optional=True),
+            requirements.ListRequirement(name="pids",
+                                         description="Filter results by process IDs. "
+                                                     "It takes the root PID namespace identifiers.",
+                                         element_type=int,
+                                         optional=True),
+            requirements.IntRequirement(name="netns",
+                                        description="Filter results by network namespace. "
+                                                    "Otherwise, all of them are shown.",
+                                        optional=True),
+        ]
+
+    @classmethod
+    def list_sockets(cls,
+                     context: interfaces.context.ContextInterface,
+                     vmlinux_module_name: str,
+                     filter_func: Callable[[int], bool] = lambda _: False):
+        """
+        Returns every single socket descriptors
+        """
+        vmlinux = context.modules[vmlinux_module_name]
+
+        sfop_addr = vmlinux.object_from_symbol("socket_file_ops").vol.offset
+        dfop_addr = vmlinux.object_from_symbol("sockfs_dentry_operations").vol.offset
+
+        fd_generator = lsof.Lsof.list_fds(context, vmlinux.name, filter_func)
+        for _pid, _task_comm, task, fd_fields in fd_generator:
+            fd_num, filp, _full_path = fd_fields
+
+            if filp.f_op not in (sfop_addr, dfop_addr):
+                continue
+
+            dentry = filp.get_dentry()
+            if not dentry:
+                continue
+
+            d_inode = dentry.d_inode
+            if not d_inode:
+                continue
+
+            socket_alloc = linux.LinuxUtilities.container_of(d_inode, "socket_alloc", "vfs_inode", vmlinux)
+            _socket = socket_alloc.socket
```

Comment:
Since `_socket` is used two lines down, this shouldn't start with an underscore.  Variables that are used in python should not start with an underscore, because python will treat them differently and it might make for weird bugs later on...

Reply:
Good catch! It was to avoid variable name collision with the socket module import which I was previously using, but then I moved that to extensions/__init__

Reply:
If you still want to avoid name collisions, I'd give it a longer name, prepending it with _ has special meaning python and tells it that you're not going to use the value stored there...